modules/Sources/UIComponents/Extensions/CustomNavigationButton.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  File.swift
+//  
+//
+//  Created by Praveen kumar Vattipalli on 02/09/23.
+//
+
+import Foundation
+import SwiftUI
+
+struct CustomNavigationButton: ViewModifier {
+    @Environment(\.dismiss) private var dismiss
+    func body(content: Content) -> some View {
+        content
+            .navigationBarBackButtonHidden(true)
+            .toolbar {
+                ToolbarItem(placement: .navigationBarLeading) {
+                    Button {
+                        dismiss()
+                    } label: {
+                        HStack {
+                            Image(systemName: "arrow.left")
+                                .foregroundColor(.black)
+                            Text("Back")
+                                .foregroundColor(.black)
+                        }
+                    }
+                }
+            }
+    }
+}
```

Comment:
Generally speaking this approach is ok, with custom button accessing @Environment. What I see as a downside of this modifier is there is no way to connect it to the TCA. If the button is is custom one, directly used in the SwiftUI's body, there could be:

```swift
Button {
     viewStore.send(...) // this is new
     dismiss()
}
```

So we either need a custom button directly used or pass generic viewStore to this modifier.